vassal-app/src/main/java/VASSAL/counters/Stack.java

```diff
@@ -771,6 +771,10 @@ public Object getLocalizedProperty(Object key) {
   @Override
   public void setMap(Map map) {
     this.map = map;
+    // And so do all our children
+    for (int i = 0; i < getPieceCount(); i++) {
```

Comment:
could you do `for (final GamePiece piece : contents)`? Or is that not allowed with arrays?

Reply:
Oh, no you actually couldn't, because `contents` is an abomination. I guess you could do `for (final GamePiece piece : asList())` but maybe that doesn't actually make anything clearer :)

Reply:
No you can't, because the array size can be larger than getPieceCount()